extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTaskClient.java

```diff
@@ -128,97 +184,251 @@ public void resume(String id)
-              "Still waiting for task [%s] to pause; will try again in [%s]", id, new Duration(sleepTime).toString()
+              "Still waiting for task [%s] to pause; will try again in [%s]",
+              id,
+              new Duration(sleepTime).toString()
           );
           Thread.sleep(sleepTime);
         }
       }
     }
+    catch (NoTaskLocationException e) {
+      return null;
+    }
     catch (IOException | InterruptedException e) {
       throw Throwables.propagate(e);
     }
   }
 
-  public KafkaIndexTask.Status getStatus(String id)
+  public KafkaIndexTask.Status getStatus(final String id)
   {
+    log.debug("GetStatus task[%s]", id);
+
     try {
       final FullResponseHolder response = submitRequest(id, HttpMethod.GET, "status", null, true);
       return jsonMapper.readValue(response.getContent(), KafkaIndexTask.Status.class);
     }
+    catch (NoTaskLocationException e) {
+      return KafkaIndexTask.Status.NOT_STARTED;
+    }
     catch (IOException e) {
       throw Throwables.propagate(e);
     }
   }
 
-  public DateTime getStartTime(String id)
+  public DateTime getStartTime(final String id)
   {
+    log.debug("GetStartTime task[%s]", id);
+
     try {
       final FullResponseHolder response = submitRequest(id, HttpMethod.GET, "time/start", null, true);
       return response.getContent() == null || response.getContent().isEmpty()
              ? null
              : jsonMapper.readValue(response.getContent(), DateTime.class);
     }
+    catch (NoTaskLocationException e) {
+      return null;
+    }
     catch (IOException e) {
       throw Throwables.propagate(e);
     }
   }
 
-  public Map<Integer, Long> getCurrentOffsets(String id, boolean retry)
+  public Map<Integer, Long> getCurrentOffsets(final String id, final boolean retry)
   {
+    log.debug("GetCurrentOffsets task[%s] retry[%s]", id, retry);
+
     try {
       final FullResponseHolder response = submitRequest(id, HttpMethod.GET, "offsets/current", null, retry);
       return jsonMapper.readValue(response.getContent(), new TypeReference<Map<Integer, Long>>() {});
     }
+    catch (NoTaskLocationException e) {
+      return ImmutableMap.of();
+    }
     catch (IOException e) {
       throw Throwables.propagate(e);
     }
   }
 
-  public Map<Integer, Long> getEndOffsets(String id)
+  public Map<Integer, Long> getEndOffsets(final String id)
   {
+    log.debug("GetEndOffsets task[%s]", id);
+
     try {
       final FullResponseHolder response = submitRequest(id, HttpMethod.GET, "offsets/end", null, true);
       return jsonMapper.readValue(response.getContent(), new TypeReference<Map<Integer, Long>>() {});
     }
+    catch (NoTaskLocationException e) {
+      return ImmutableMap.of();
+    }
     catch (IOException e) {
       throw Throwables.propagate(e);
     }
   }
 
-  public void setEndOffsets(String id, Map<Integer, Long> endOffsets)
+  public Boolean setEndOffsets(final String id, final Map<Integer, Long> endOffsets)
```

Comment:
Ah, I guess you're doing this so you can stuff it in a future? Even so, it'd be nice to have this be `boolean` if that compiles properly. The return value of "Boolean" makes me wonder "Why does this need to be nullable"